MetaMask/metamask-extension
# feat: import tokens redesign

## **Description**

This ticket aims to redesign the import token flow.
Here is the ticket: https://consensyssoftware.atlassian.net/browse/MMASSETS-62
And Figma: https://www.figma.com/file/TfVzSMJA8KwpWX8TTWQ2iO/MMASSETS-15%3A-Asset-list-and-details?type=design&node-id=1491-56878&mode=design

Notes: 
1- In the Figma designs you will notice that the notification you get when you import a token is a black box notif; but the one in this PR is a green box; we have decided to eventually go with a notification component that already existed.
2- In the Figma design (2.0A) the confirmation page should also show the balance in Fiat. We decided to split that part in a separate ticket and for this PR use the same tokenBalance component used in the old confirmation modal.

## **Related issues**

Fixes: #https://github.com/MetaMask/metamask-extension/issues/21052

## **Manual testing steps**

1. Go to home page
2. Click on import tokens button
3. try searching for your token
4. Select the tokens you need
5. Click Next
6. If you want to exit at this point, click the "x" and confirm; and if you want to go back click "back"
7. If you clicked back on step 6; click import and you should be redirected to home page.
8. You should be able to see your tokens on home page.

## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-extension/assets/10994169/858b92aa-0ad4-48d1-b71f-b8ec90f62124

### **After**

https://github.com/MetaMask/metamask-extension/assets/10994169/0a5f8d4d-39c0-45fb-a76e-35f278e9070a


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
